Tests/UnitTests/Core/EventData/ParameterSetTests.cpp

```diff
@@ -638,5 +811,464 @@ BOOST_AUTO_TEST_CASE(parset_parID_mapping) {
+      std::nullopt, 0, 1.7, -0.15);
+  BOOST_CHECK(assigned != other);
+  assigned = other;
+  BOOST_CHECK(assigned == other);
+
+  // check move assignment
+  BOOST_CHECK(first != assigned);
+  first = ParameterSet<FreeParametersIndices, eFreePos0, eFreePos1, eFreePos2>(
+      assigned);
+  BOOST_CHECK(first == assigned);
+
+  // check swap method
+  ParameterSet<FreeParametersIndices, eFreePos0, eFreePos1, eFreePos2> lhs(
+      cov, x, y, z);
+  ParameterSet<FreeParametersIndices, eFreePos0, eFreePos1, eFreePos2> rhs(
+      std::nullopt, 2 * x, 2 * y, 2 * z);
+  ParameterSet<FreeParametersIndices, eFreePos0, eFreePos1, eFreePos2>
+      lhs_copy = lhs;
+  ParameterSet<FreeParametersIndices, eFreePos0, eFreePos1, eFreePos2>
+      rhs_copy = rhs;
+
+  BOOST_CHECK(lhs != rhs && lhs == lhs_copy && rhs == rhs_copy);
+  using std::swap;
+  swap(lhs, rhs);
+  BOOST_CHECK(lhs != rhs && rhs == lhs_copy && lhs == rhs_copy);
+}
+
+/**
+ *  @brief Unit test for comparison operators in ParameterSet
+ *
+ *  @sa ParameterSet::operator==, ParameterSet::operator!=
+ */
+BOOST_AUTO_TEST_CASE(free_parset_comparison_tests) {
+  // covariance matrix
+  ActsSymMatrixD<3> cov;
+  cov << 1, 0, 0, 0, 1.2, 0.2, 0, 0.2, 0.7;
+
+  // parameter values
+  double x = 0.5;
+  double y = -0.2;
+  double z = 0.3;
+
+  // parameter set with covariance matrix
+  ParameterSet<FreeParametersIndices, eFreePos0, eFreePos1, eFreePos2> first(
+      cov, x, y, z);
+  ParameterSet<FreeParametersIndices, eFreePos0, eFreePos1, eFreePos2> second(
+      std::nullopt, 2 * x, 2 * y, 2 * z);
+
+  // check self comparison
+  BOOST_CHECK(first == first);
+  BOOST_CHECK(not(first != first));
+
+  // check mutual exclusivity
+  BOOST_CHECK(first != second);
+  BOOST_CHECK(not(first == second));
+  first = second;
+  BOOST_CHECK(first == second);
+
+  // check that comparison fails for unequal parameter values
+  second.setParameter<eFreePos0>(3 * x);
+  BOOST_CHECK(first != second);
+  first = second;
+  BOOST_CHECK(first == second);
+
+  second.setParameter<eFreePos1>(3 * y);
+  BOOST_CHECK(first != second);
+  first = second;
+  BOOST_CHECK(first == second);
+
+  second.setParameter<eFreePos2>(3 * z);
+  BOOST_CHECK(first != second);
+  first = second;
+  BOOST_CHECK(first == second);
+
+  // check that comparison fails for unequal covariance matrices
+  second.setCovariance(cov);
+  BOOST_CHECK(first != second);
+  first = second;
+  BOOST_CHECK(first == second);
+
+  cov(0, 0) = 2 * cov(0, 0);
+  second.setCovariance(cov);
+  BOOST_CHECK(first != second);
+  first = second;
+  BOOST_CHECK(first == second);
+}
+
+/**
+ * @brief Unit test for projection matrices in ParameterSet
+ *
+ * Checks the correctness of the projection matrices from the full parameter
+ * space
+ * onto different parameter sub-spaces
+ *
+ * @sa ParameterSet::projector
+ */
+BOOST_AUTO_TEST_CASE(free_parset_projection_tests) {
+  ActsMatrixD<1, eFreeParametersSize> z_proj;
+  z_proj << 0, 0, 1, 0, 0, 0, 0, 0;
```

Comment:
You could consider disabling clang format here to be able to align the matrix rows so that’s they become more readable. Up to you though.

Reply:
Did it, looks better